src/Shared/validations.tsx

```diff
@@ -47,22 +60,67 @@ export const validateRequiredPositiveNumber = (value: string | number): Validati
     if (!value) {
         return {
             isValid: false,
-            message: 'This field is required',
+            message: MESSAGES.PROVIDE_A_NUMBER,
         }
     }
 
     // TODO: Look if regex is available for this, since 0002 is not a valid number
     if (!/^\d+(\.\d+)?$/.test(value.toString())) {
         return {
             isValid: false,
-            message: 'This field should be a valid positive number',
+            message: MESSAGES.VALID_POSITIVE_NUMBER,
+        }
+    }
+
+    const numericValue = Number(value)
+
+    if (numericValue > Number.MAX_SAFE_INTEGER) {
+        return {
+            isValid: false,
+            message: MESSAGES.MAX_SAFE_INTEGER,
+        }
+    }
+
+    if (Number(value) <= 0) {
+        return {
+            isValid: false,
+            message: MESSAGES.VALID_POSITIVE_NUMBER,
+        }
+    }
+
+    return {
+        isValid: true,
+    }
+}
+
+export const validatePositiveInteger = (value: string | number): ValidationResponseType => {
```

Comment:
this is duplicated in the above function as well. Can we reuse this function in the above one?

Reply:
Messaging and requirements are different